PlasmaMVP/throughputOptimization.md

```diff
@@ -0,0 +1,52 @@
+### Increase Throughput of Plasma
+
+Rather than submitting a block to the Ethereum rootchain and waiting for presumed finality before submitting the next block; we propose changes to the submitBlock function that allow blocks to be added before the previous blocks get finalized. We effectively allow the blocks to be added "asynchronously".
+
+The submitBlock function changes to:
+``submitBlock(blockNum, root, prevRoot, sig)``
```

Comment:
It would be good to finalize the spec before merging this. prevRoot is most likely not to be needed. We can grab the hash of the current root but indexing [blockNum+1]

Reply:
We need the previous root to serve as a "commitment" by the validator of past history so that no one can change history in the case that a middle block gets reorged.

Reply:
you are right. looks good